test/src/us/kbase/kbasefeaturevalues/test/KBaseFeatureValuesServerTest.java

```diff
@@ -195,61 +197,111 @@ public void testMainPipeline() throws Exception {
+        comparing the two fields that are populated: best K and K clusters
+        and empty additionalProperties */
+        Assert.assertNotNull("k exists ", kLoad);
+        Assert.assertNotNull("c size exists ", estKResLoad.getEstimateClusterSizes().size());
+        Assert.assertEquals(3, kLoad);
+        Assert.assertEquals(5, estKResLoad.getEstimateClusterSizes().size());
+        Map<String, Object> additionalPropertiesLoad = new HashMap<String, Object>();
+        Assert.assertEquals(additionalPropertiesLoad, estKRes.getAdditionalProperties());
+        for (int i = 0; i < estKResLoad.getEstimateClusterSizes().size(); i++) {
+            Tuple2<Long, Double> item = estKResLoad.getEstimateClusterSizes().get(i);
+            Assert.assertEquals(2L + i, (long) item.getE1());
+            Assert.assertTrue((double) item.getE2() > 0);
+        }
+
+
+        /////////////// estimate K new /////////////////
+        EstimateKResult estKResNew = impl.estimateKNew(new EstimateKParamsNew().withInputMatrix(testWsName + "/" +
                 exprObjName).withRandomSeed(123L).withOutWorkspace(testWsName)
                 .withOutEstimateResult(estimNewObjName), token, getContext());
-        ObjectData res1new = wscl.getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
-                .withName(estimNewObjName))).get(0);
-        EstimateKResult estKResNew = res1new.getData().asClassInstance(EstimateKResult.class);
+
         long kNew = estKResNew.getBestK();
-        Assert.assertEquals(k, kNew);
-        //System.out.println("Cluster count qualities: " + estKResNew.getEstimateClusterSizes());
+
+        /*
+        this is testing the object returned by the method
+        comparing the two fields that are populated: best K and K clusters
+        and empty additionalProperties */
+        Assert.assertEquals(3, kNew);
         Assert.assertEquals(estKRes.getEstimateClusterSizes().size(), estKResNew.getEstimateClusterSizes().size());
         for (int i = 0; i < estKResNew.getEstimateClusterSizes().size(); i++) {
             Tuple2<Long, Double> entry = estKRes.getEstimateClusterSizes().get(i);
             Tuple2<Long, Double> entryNew = estKResNew.getEstimateClusterSizes().get(i);
-            Assert.assertEquals((long)entry.getE1(), (long)entryNew.getE1());
-            Assert.assertEquals((double)entry.getE2(), (double)entryNew.getE2(), 1e-10);
+            Assert.assertEquals((long) entry.getE1(), (long) entryNew.getE1());
+            Assert.assertEquals((double) entry.getE2(), (double) entryNew.getE2(), 1e-10);
+        }
+
+        //loading saved result object
+        ObjectData res1New = getWsClient().getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
+                .withName(estimObjName))).get(0);
+        EstimateKResult estKNewResLoad = res1New.getData().asClassInstance(EstimateKResult.class);
+        long kNewLoad = estKNewResLoad.getBestK();
```

Comment:
It seems like there could be some DRYing up of these repeated test methods in the future